### PR TITLE
Feature/45 overview asset widgets

### LINKED ISSUE #45: (Overview) - My Assets Widget
A Quick way to show the Assets I own/manage/consume.

### PR DESCRIPTION
**Pull request by IAMLEGION98**
Developing the **Overview Page with Widgets** displaying info about assets owned by the user.
Visualizations and charts for the many use-cases (role-specific) as mentioned in #45 

closes #45

### CODE REVIEW

File: client/src/js/overview/overview.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+              </div>
+            );
+          }
+        })()}
+      </main>
+    </>
+  );
+};
+const mapStateToProps = (state) => ({
+  role: state.role,
```
Comment (atapas) [inline client/src/js/overview/overview.js:111]: Why redux?

Moreover this is not the good way to do redux.

Comment (IAMLEGION98) [reply client/src/js/overview/overview.js:111]: The Redux state is for the role based permission to function . Could you elaborate Sir?

Comment (atapas) [reply client/src/js/overview/overview.js:111]: https://blog.greenroots.info/step-by-step-guide-blend-redux-with-reactjs-ck1tixqp00055avs100ncho2x

File: client/src/js/overview/widgets/Widget2.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import DevicesIcon from "@material-ui/icons/Devices";
+import { useSelector, useDispatch } from "react-redux";
+
+import StopIcon from "@material-ui/icons/Stop";
+import { Card, Col, Container, Row, Badge } from "react-bootstrap";
+import Piechart from "../charts/Piechart";
+import SettingsApplicationsIcon from "@material-ui/icons/SettingsApplications";
+import AppsIcon from "@material-ui/icons/Apps";
+import { getAllAssets } from "../../services/AssetsInfoFetch";
+import { useHistory } from "react-router-dom";
```
Comment (atapas) [inline client/src/js/overview/widgets/Widget2.js:11]: What is widget2 for?

Comment (VarshaRavichandran) [reply client/src/js/overview/widgets/Widget2.js:11]: Just to test out the positioning sir..

Comment (VarshaRavichandran) [reply client/src/js/overview/widgets/Widget2.js:11]: Once we start working on the nnext component we will be modifying the widget2 (including the filename)

**Comment by Jayalakshmi02** [comment]
Asset view page for software assets
Screenshot (279)
Asset view page for hardware assets with server tag
Screenshot (291)
Asset view page for hardware assets that are not servers
Screenshot (280)
Form validation for adding new user
Screenshot (284)
Form validation for adding new asset
Screenshot (285)
Clickable hardware sector on widget (lab admin login)
Screenshot (270)
On click of hardware portion of chart in widget, redirection to asset list showing the hardware assets owned by lab admin
Screenshot (276)
Clickable software sector on widget (lab admin login)
Screenshot (271)
On click of software portion of chart in widget, redirection to asset list showing the software assets owned by lab admin
Screenshot (275)
Asset list displayed when lab admin clicks on total in widget
Screenshot (293)
Clickable software number tile in widget
Untitled
Asset list displaying software assets owned by labadmin on clicking software number tile
Screenshot (295)
My asset widget on labowner login
Screenshot (281)
On click of hardware portion of chart in widget, redirection to asset list showing the hardware assets owned by lab owner
Screenshot (282)
Widget showing asset consumed by consumer
Screenshot (287)
Asset list displayed on clicking pie chart sector by consumer
Screenshot (288)

**Comment by Jayalakshmi02** [comment]
The input field for contract end date while adding new assets should be calender dropdown as the user shouldn't have to calculate date in bigint

**Pull request merged**